runtime/vm/vmprops.c

```diff
@@ -572,35 +574,45 @@ initializeSystemProperties(J9JavaVM * vm)
 		}
 	}
 
-#if JAVA_SPEC_VERSION < 12
-	/* Following system properties are defined via java.lang.VersionProps.init(systemProperties) and following settings within System.ensureProperties() */
-	{
-		const char *classVersion = NULL;
-		const char *specificationVersion = NULL;
-
-		/* Properties that always exist */
-		switch (j2seVersion) {
-		case J2SE_18:
-			classVersion = "52.0";
-			specificationVersion = "1.8";
-			break;
-		case J2SE_V11:
-		default:
-			classVersion = "55.0";
-			specificationVersion = "11";
-			break;
-		}
-		rc = addSystemProperty(vm, "java.class.version", classVersion, 0);
-		if (J9SYSPROP_ERROR_NONE != rc) {
+	if (JAVA_SPEC_VERSION == 8) {
+		specificationVersion = "1.8";
+	} else {
+		bufLen = j9str_printf(PORTLIB, NULL, 0, "%d", JAVA_SPEC_VERSION);
+		if (bufLen > 0) {
+			specificationVersion = j9mem_allocate_memory(bufLen, OMRMEM_CATEGORY_VM);
+			if (NULL != specificationVersion) {
+				j9str_printf(PORTLIB, specificationVersion, bufLen, "%d", JAVA_SPEC_VERSION);
+			} else {
+				rc = J9SYSPROP_ERROR_OUT_OF_MEMORY;
+				goto fail;
+			}
+		} else {
+			rc = J9SYSPROP_ERROR_EMPTY_JAVA_SPEC_VERSION;
 			goto fail;
 		}
-
+	}
+	if (JAVA_SPEC_VERSION < 12) {
+		/* System property "java.specification.version" is defined via java.lang.VersionProps.init(systemProperties) within System.ensureProperties() */
 		rc = addSystemProperty(vm, "java.specification.version", specificationVersion, 0);
 		if (J9SYSPROP_ERROR_NONE != rc) {
 			goto fail;
 		}
-
-		rc = addSystemProperty(vm, "java.vm.specification.version", specificationVersion, 0);
+	}
+	rc = addSystemProperty(vm, "java.vm.specification.version", specificationVersion, 0);
+	if (J9SYSPROP_ERROR_NONE != rc) {
+		goto fail;
+	}
+	
+	if (JAVA_SPEC_VERSION < 12) {
+		/* Following system properties are defined via java.lang.VersionProps.init(systemProperties) within System.ensureProperties() */
+		const char *classVersion = NULL;
+		if (JAVA_SPEC_VERSION == 8) {
+			classVersion = "52.0";
+		} else {
+			/* Java 11 */
+			classVersion = "55.0";
```

Comment:
This should handle Java 12, 13, etc.

Reply:
`JDK 12+` class versions are set by `java.lang.VersionProps.init(systemProperties)` within `System.ensureProperties()`.

Reply:
Since it used to be set in the VM, I think we should be adding this for Java 12 and up, even though VersionProps.init() sets it later. Then the value is available in JVMTI.

Reply:
Note it can be done automatically as 44 + JAVA_SPEC_VERSION

Reply:
Class version technically is owned by VM, set it in native for all Java versions. Please review again.